Assets/Tests/PlayModeTests/SessionTests.cs

```diff
@@ -0,0 +1,342 @@
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+            // RQ will have Consent change request and Session begin request
+            Assert.AreEqual(2, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.RemoveConsent(new Consents[] { Consents.Sessions });
+            //RQ will have consent change request and session begin request
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+            // RQ will have Consent change request and Session begin request
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(2, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+
+        }
+
+        /// <summary>
+        /// It validates the request of 'ExtendSessionAsync'.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestSessionExtend()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+            };
+
+            System.DateTime sessionStartTime = System.DateTime.Now;
+
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+
+            CountlyRequestModel requestModel = Countly.Instance.Session._requestCountlyHelper._requestRepo.Dequeue();
+            string myUri = requestModel.RequestUrl;
+            NameValueCollection values = HttpUtility.ParseQueryString(myUri);
+
+            Assert.AreEqual("1", values.Get("begin_session"));
+            Assert.IsNotNull(values.Get("metrics"));
+
+            System.DateTime startTime = System.DateTime.UtcNow;
+            do {
+                yield return null;
+            }
+            while ((System.DateTime.UtcNow - startTime).TotalSeconds < 2.0);
+
+            Countly.Instance.Session.ExtendSessionAsync();
+
+            startTime = System.DateTime.UtcNow;
+            do {
+                yield return null;
+            }
+            while ((System.DateTime.UtcNow - startTime).TotalSeconds < 2.0);
+
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+
+            requestModel = Countly.Instance.Session._requestCountlyHelper._requestRepo.Dequeue();
+            myUri = requestModel.RequestUrl;
+            values = HttpUtility.ParseQueryString(myUri);
+
+            double duration = (System.DateTime.Now - sessionStartTime).TotalSeconds;
```

Comment:
should not use "System.DateTime.Now" but it should use "startTime " or any similar timestamp right after "ExtendSessionAsync"